### PR TITLE
Support a mask2k file to prioritize sections

### PR DESCRIPTION
**Pull request by octylFractal**
Please set mask: true if your template supports this. Already set for `mlp` as we uploaded one previously.

### CODE REVIEW

File: minimap.impl.user.js [hunk 1]
```diff
@@ -43,32 +43,27 @@ const { html, render } = mlp_uhtml;
...
-      return `https://media.githubusercontent.com/media/r-ainbowroad/minimap/d/main/${templateName}/canvas2k.png`;
-    }
-    return `https://raw.githubusercontent.com/r-ainbowroad/minimap/d/main/${templateName}/canvas2k.png`;
-  };
-  const getRPlaceTemplateBotUrl = function (templateName) {
-    if (rPlaceTemplatesGithubLfs) {
-      return `https://media.githubusercontent.com/media/r-ainbowroad/minimap/d/main/${templateName}/bot2k.png`;
-    }
-    return `https://raw.githubusercontent.com/r-ainbowroad/minimap/d/main/${templateName}/canvas2k.png`;
+  const rPlaceTemplatesBaseUrl = rPlaceTemplatesGithubLfs
```
Comment (bb010g) [inline minimap.impl.user.js:46]: nit: `rPlaceTemplateBaseUrl`

File: minimap.impl.user.js [hunk 2]
```diff
@@ -418,22 +440,109 @@ const { html, render } = mlp_uhtml;
...
+  }
+
+  /**
+   * Select a random pixel.
+   *
+   * @param {[number, number][]} diff
+   * @return {[number, number]}
+   */
+  function selectRandomPixel(diff) {
+    if (typeof rPlaceTemplate.maskUrl === "undefined" || typeof rPlaceMask === "undefined") {
```
Comment (bb010g) [inline minimap.impl.user.js:539]: nit: `rPlaceTemplate.maskUrl === undefined || rPlaceMask === undefined`

JavaScript equality operators can't be overridden, so this is preferrable.

File: minimap.impl.user.js [hunk 3]
```diff
@@ -574,8 +683,7 @@ const { html, render } = mlp_uhtml;
         .shadowRoot.querySelector("mona-lisa-color-picker")
         .shadowRoot.querySelector("button.confirm");
       if (placeButton && diff.length > 0) {
-        const randID = Math.floor(Math.random() * diff.length);
-        const randPixel = diff[randID];
+        const randPixel = selectRandomPixel(diff);
```
Comment (bb010g) [inline minimap.impl.user.js:686]: If you could refactor `selectRandomPixel(diff)` to return an object with `.x` and `.y` properties and then use those below, that'd be lovely.

Alternatively, have it destructure out some `x` and `y` bindings like you do above.

File: minimap.impl.user.js [hunk 4]
```diff
@@ -418,22 +440,109 @@ const { html, render } = mlp_uhtml;
       rPlaceTemplate.botUrl !== undefined && settings.getSetting("bot").enabled
         ? rPlaceTemplate.botUrl
         : rPlaceTemplate.canvasUrl;
-    setTimeout(restoreBotWorkingRightNow, 10 * 1000);
```
Comment (bb010g) [inline minimap.impl.user.js:421]: This failsafe restore should be kept, unless you know that we are going to always restore it.

File: minimap.impl.user.js [hunk 5]
```diff
@@ -404,6 +404,28 @@ const { html, render } = mlp_uhtml;
...
+          resolve(new Uint8Array(res.response));
+        },
+        onerror: function (res) {
+          reject(res);
+        }
+      });
+    })
+  }
+
+  function createPngDataUrlForBytes(bytes) {
```
Comment (bb010g) [inline minimap.impl.user.js:425]: `getPngDataUrlForBytes`

**Review by bb010g — CHANGES_REQUESTED** [review]

**Comment by bb010g** [comment]
You also might wanna document the mask format in the README, at the bottom.

**Comment by octylFractal** [comment]
Closed in favor of #10.

**Pull request closed without merging**